toolkit/tools/pkg/imagecustomizerlib/customizeverity.go

```diff
@@ -0,0 +1,224 @@
+	}
+
+	// Extract the version from the kernel filename (e.g., vmlinuz-5.15.131.1-2.cm2 -> 5.15.131.1-2.cm2)
+	kernelVersion := strings.TrimPrefix(kernelFiles[0], "vmlinuz-")
+
+	err = imageChroot.Run(func() error {
+		// TODO: Config Dracut module systemd-veritysetup - task 6421.
+		err = shell.ExecuteLiveWithErr(1, "dracut", "-f", "--kver", kernelVersion, "-a", dracutModuleName)
+		return err
+	})
+	if err != nil {
+		return fmt.Errorf("failed to build dracut module - (%s):\n%w", dracutModuleName, err)
+	}
+
+	return nil
+}
+
+func updateMarinerCfgWithInitramfs(imageChroot *safechroot.Chroot) error {
+	var err error
+
+	initramfsPath := filepath.Join(imageChroot.RootDir(), "boot/initramfs-*")
+	// Fetch the initramfs file name.
+	var initramfsFiles []string
+	initramfsFiles, err = filepath.Glob(initramfsPath)
+	if err != nil {
+		return fmt.Errorf("failed to list initramfs file: %w", err)
+	}
+
+	// Ensure an initramfs file is found
+	if len(initramfsFiles) != 1 {
+		return fmt.Errorf("expected one initramfs file, but found %d", len(initramfsFiles))
+	}
+
+	newInitramfs := filepath.Base(initramfsFiles[0])
+
+	cfgPath := filepath.Join(imageChroot.RootDir(), "boot/mariner.cfg")
+
+	lines, err := file.ReadLines(cfgPath)
+	if err != nil {
+		return fmt.Errorf("failed to read mariner.cfg: %w", err)
+	}
+
+	// Update lines to reference the new initramfs
+	for i, line := range lines {
+		if strings.HasPrefix(line, "mariner_initrd=") {
+			lines[i] = "mariner_initrd=" + newInitramfs
+		}
+	}
+	// Write the updated lines back to mariner.cfg using the internal method
+	err = file.WriteLines(lines, cfgPath)
+	if err != nil {
+		return fmt.Errorf("failed to write to mariner.cfg: %w", err)
+	}
+
+	return nil
+}
+
+func updateGrubConfig(dataPartitionIdType imagecustomizerapi.IdType, dataPartitionId string,
+	hashPartitionIdType imagecustomizerapi.IdType, hashPartitionId string, rootHash string, grubCfgFullPath string,
+) error {
+	var err error
+
+	// Format the dataPartitionId and hashPartitionId using the helper function.
+	formattedDataPartition, err := systemdFormatPartitionId(dataPartitionIdType, dataPartitionId)
+	if err != nil {
+		return err
+	}
+	formattedHashPartition, err := systemdFormatPartitionId(hashPartitionIdType, hashPartitionId)
+	if err != nil {
+		return err
+	}
+
+	newArgs := fmt.Sprintf(
+		"rd.systemd.verity=1 roothash=%s systemd.verity_root_data=%s systemd.verity_root_hash=%s systemd.verity_root_options=panic-on-corruption",
+		rootHash, formattedDataPartition, formattedHashPartition,
+	)
+
+	// Read grub.cfg using the internal method
+	lines, err := file.ReadLines(grubCfgFullPath)
+	if err != nil {
+		return fmt.Errorf("failed to read grub config: %v", err)
+	}
+
+	var updatedLines []string
+	for _, line := range lines {
+		trimmedLine := strings.TrimSpace(line)
+		if strings.HasPrefix(trimmedLine, "linux ") {
+			// Append new arguments to the line that starts with "linux"
+			line += " " + newArgs
```

Comment:
This doesn't seem to be safe for multiple runs of imagecustomizer, in that it would blindly append newArgs even though the args already existed. I would find a regex search and replace more resilient. That way you can be sure there aren't unintentionally two values of, say, roothash, on the command line.

Reply:
Thanks, fixed.